lib/driver.js

```diff
@@ -382,7 +380,7 @@ class XCUITestDriver extends BaseDriver {
   }
 
   async startWda (sessionId, realDevice) {
-    this.wda = new WebDriverAgent(this.xcodeVersion, this.opts);
+    this.wda = new WebDriverAgent(this.xcodeVersion, Object.assign(this.opts, {isFirstSession: this.isFirstSession}));
```

Comment:
Yes. That's why I interpreted @imurchie as saying we should probably change to something like:

```js
Object.assign({}, this.opts, {isFirstSession: this.isFirstSession})
```

so that a new object is created.